File: controller/Reward.cpp
//Reward.cpp
#include "Reward.h"

bool isPriming = false;
unsigned long rewardDurationMs = 1000; 

void setupReward() {
    //pinMode(M2, OUTPUT);
    //digitalWrite(M2, LOW); 

    // PWM configuration
    ledcAttachPin(E2, PUMP_PWM_CHANNEL);
    ledcSetup(PUMP_PWM_CHANNEL, FREQ, RESOLUTION);
}

void dispenseReward() {
    ledcWrite(PUMP_PWM_CHANNEL, 255);
}

void stopRewardDispense() {
    ledcWrite(PUMP_PWM_CHANNEL, 0);
}

void primeFeedingTube() {
    isPriming = true;
    unsigned long startTime = millis();
    while (isPriming && millis() - startTime < 120000) { 
        ledcWrite(PUMP_PWM_CHANNEL, 255); 
        
        if (Serial.available() > 0) {
            char command = Serial.read();
            if (command == 'x') {
                stopPriming();
                break;
            }
        }
        delay(100); 
    }
    ledcWrite(PUMP_PWM_CHANNEL, 0); 
    if (isPriming) { 
        Serial.println("Priming Finished");
    }
}

void stopPriming() {
    ledcWrite(PUMP_PWM_CHANNEL, 0);
    isPriming = false;
    Serial.println("Priming stopped");
}

File: _archive/src/peripherals/reward.py
import pigpio
import time

class Reward:
    def __init__(self, pi, pin):
        self.pi = pi
        self.pin = pin
        self.is_priming = False
        self.reward_duration_ms = 1000

    def setup_reward(self):
        self.pi.set_mode(self.pin, pigpio.OUTPUT)
        self.pi.set_PWM_dutycycle(self.pin, 0)
        self.pi.set_PWM_range(self.pin, 255)
        self.pi.set_PWM_frequency(self.pin, 5000)  # Match frequency from Arduino code

    def dispense_reward(self):
        self.pi.set_PWM_dutycycle(self.pin, 255)

    def stop_reward_dispense(self):
        self.pi.set_PWM_dutycycle(self.pin, 0)

    def prime_feeding_tube(self):
        self.is_priming = True
        start_time = time.time()

        while self.is_priming and (time.time() - start_time) < 120:
            self.pi.set_PWM_dutycycle(self.pin, 255)  # Full power

            command = input("Enter 'x' to stop priming: ")
            if command == 'x':
                self.stop_priming()
                break

            time.sleep(0.1) 

        self.pi.set_PWM_dutycycle(self.pin, 0)
        if self.is_priming:
            print("Priming Finished")

    def stop_priming(self):
        self.pi.set_PWM_dutycycle(self.pin, 0)
        self.is_priming = False
        print("Priming stopped")
